### DISCUSSION TITLE
[ext:joomla] How to display content from a specific article ID

### DISCUSSION THREAD

**Discussion by raramuridesign** [joomlatools/joomlatools-pages#746] [Q&A] [answered]
How do we go about displaying information from a specific article ID?
Matthew

**Comment by johanjanssens** [comment, marked as answer]
Hello Matthew,

Displaying content from a single Joomla article is very easy in Pages using the Joomla extension. Here are the steps (for 0.21):

### Install Joomla extension

Installing the Joomla extension is really simple, just download http://files.joomlatools.com/pages@0.21.1/extension-joomla.zip and put it in `/joomlatools-pages/extensions` directory. 

### Getting an article by id or alias

The Joomla extension provides a custom`article()` template function that allows to get a single article by it's `id` or` alias`.

#### For example to retrieve the article with id = 1

```php
<article>
   <h1><?= article(1)->title ?></h1>
    <p><?= article(1)->excerpt.article(1)->text ?></p>
</article> 
```

To get an idea of all the different properties you can do 

```php
<?= var_dump(article(1)); ?>
```

- `excerpt` is what is know as the ìntrotext` in joomla
- `text`is what is know as the `fulltext` in joomla

_More info about properties and naming conventions see below._

#### For example to retrieve the article with alias = 'my-article'

```php
<article>
   <h1><?= article('my-article')->title ?></h1>
     <p><?= article('my-article')->excerpt.article('my-article')->text ?></p>
</article> 
```

### Defining custom aliases 

Using Joomla article aliases directly could results into articles not being found it the alias is edited (by accident). To resolve that and to still make it possible to reference an article by a string you can define your own custom static aliases through the [configuration](https://github.com/joomlatools/joomlatools-pages/wiki/Configuration)

#### For example define an alias called `about-us` and link it to article with id = 1

```php
<?php
return array(
    'extension_config' =>
    [
	'ext:joomla.model.articles'  =>
	[
		'aliases' => [
			'about-us' => 1
		]
	]
   ]
);
```

You can now do

```php
<article>
   <h1><?= article('about-us')->title ?></h1>
    <p><?= article('about-us')->text ?></p>
</article> 
```

### Blog example

The blog example site contains a working example of the `article()` function in the about page: 

- https://github.com/joomlatools/joomlatools-pages/blob/master/contrib/sites/blog/pages/about.html.php
- https://github.com/joomlatools/joomlatools-pages/blob/master/contrib/sites/blog/config.php#L43 (for the config)

Suggest you check out this example and try to get it runnning locally, provides a good starting point to explore and play with Pages.

For more info about the Joomla extension check the PR, which contains all the docs: https://github.com/joomlatools/joomlatools-pages/pull/363

### Internals - Implicit content model

A bit of further info that could be useful. You will notice that the article entity properties are not the same as those in Joomla. Those properties go back almost 2 decades now for Pages I'm working on unified content model for articles and other content entities. The property naming I'm using is based off other CMS systems and naming conventions.

For example the collection model `ext:joomla:model.menus` and can be inter-changeable used in the same way as the Pages collection, both use the same implicit content model. Making it very easy to create templates and partials that work for both. More info: https://github.com/joomlatools/joomlatools-pages/discussions/601

Related also check: https://github.com/joomlatools/joomlatools-pages/discussions/673

**Reply by raramuridesign** [reply]
@johanjanssens Thank you so much. Worked out of the box, so really great.

**Reply by johanjanssens** [reply]
Thanks, happy to hear that

**Reply by raramuridesign** [reply]
Adding support for images use this 

`<img src="<?= article(2)->image->url ?>" class="img-fluid" alt="sample image">`

**Reply by johanjanssens** [reply]
That works indeed, see also: https://github.com/joomlatools/joomlatools-pages/discussions/673
